script.js

```diff
@@ -0,0 +1,178 @@
+// function defs --------------------------------------------
+var ttt_array = [ [],[],[] ];
+var play = true;
+var player1;
+var player2;
+var turnCounter = 0;
+
+function Player (name, turn, token){
+  this.name = name;
+  this.turn = turn;
+  this.token = token;
+}
+
+function initPlayers(event){
+  event.preventDefault();
+  player1Name = document.getElementById("player1").value || "harry";
+  player2Name = document.getElementById("player2").value || "voldemort";
+  player1 = new Player(player1Name, true, "animals_walrus.gif");
+  player2 = new Player(player2Name, false, "animals_octop.gif");
+  document.getElementById("space").removeChild(document.getElementById("player_names"));
+  document.getElementById("winner_banner").innerText = player1.name + "\'s turn";
+
+  var topLeft = document.getElementById("top_left");
+  topLeft.addEventListener("click", addImage);
+  var topCenter = document.getElementById("top_center");
+  topCenter.addEventListener("click", addImage);
+  var topRight = document.getElementById("top_right");
+  topRight.addEventListener("click", addImage);
+  var middleLeft = document.getElementById("middle_left");
+  middleLeft.addEventListener("click", addImage);
+  var middleCenter = document.getElementById("middle_center");
+  middleCenter.addEventListener("click", addImage);
+  var middleRight = document.getElementById("middle_right");
+  middleRight.addEventListener("click", addImage);
+  var bottomLeft = document.getElementById("bottom_left");
+  bottomLeft.addEventListener("click", addImage);
+  var bottomCenter = document.getElementById("bottom_center");
+  bottomCenter.addEventListener("click", addImage);
+  var bottomRight = document.getElementById("bottom_right");
+  bottomRight.addEventListener("click", addImage);
+  var playAgain = document.getElementById("play_again");
+  playAgain.addEventListener("click", reset);
+}
+
+function updateTurn(player_current, player_other){
+  player_current.turn = false;
+  player_other.turn = true;
+  document.getElementById("winner_banner").innerText = player_other.name + "\'s turn";
+}
+
+function addImage(event) {
+  if (play){
+    var image = document.createElement("img");
+    this.appendChild(image);
+    this.removeEventListener("click", addImage);
+    if(player1.turn === true){
```

Comment:
i think this is an elegant solution to not allow someone to play on the same space.